staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/merge.go

```diff
@@ -42,31 +43,12 @@ func MergeSpecs(staticSpec *spec.Swagger, crdSpecs ...*spec.Swagger) *spec.Swagg
 	}
 
 	for _, s := range crdSpecs {
-		mergeSpec(&specToReturn, s)
-	}
-
-	return &specToReturn
-}
-
-// mergeSpec copies paths and definitions from source to dest, mutating dest, but not source.
-// We assume that conflicts do not matter.
-func mergeSpec(dest, source *spec.Swagger) {
-	if source == nil || source.Paths == nil {
-		return
-	}
-	if dest.Paths == nil {
-		dest.Paths = &spec.Paths{}
-	}
-	for k, v := range source.Definitions {
-		if dest.Definitions == nil {
-			dest.Definitions = spec.Definitions{}
-		}
-		dest.Definitions[k] = v
-	}
-	for k, v := range source.Paths.Paths {
-		if dest.Paths.Paths == nil {
-			dest.Paths.Paths = map[string]spec.PathItem{}
+		// Path conflicts are ignored by keeping the existing paths. I.e. the
+		// first spec with highest priority wins its paths.
+		if err := aggregator.MergeSpecsIgnorePathConflict(&specToReturn, s); err != nil {
```

Comment:
this will grow quadratically if I am not mistaken. We should not call the full merge.